### ISSUE TITLE
Saving the buffer twice…

### ISSUE THREAD

**Issue by ahmedelgabri** [mhartington/formatter.nvim#17]
I have an issue that started to happen after #12, most of the time I need to save the buffer twice. The first will run the formatter as expected but then because it updates the buffer with the new formatted content, the buffer becomes modified again & I have to save it again to be saved with the formatted content. 

I'm still trying to debug & see if that has anything to do with my setup (very unlikely because I didn't change much after I modified [my config](https://github.com/ahmedelgabri/dotfiles/commits/master) to match the changes in #12)

This is my [`formatter`](https://github.com/ahmedelgabri/dotfiles/blob/2d88b4762edffeea23ea9bb8ff9a6214a1d9ecf8/roles/vim/files/.vim/lua/_/plugins.lua#L176-L292) config.

**Comment by mhartington** [comment]
How are you triggering the formatting on save?

**Comment by mhartington** [comment]
Looking at your config, you're using just `Format`

https://github.com/ahmedelgabri/dotfiles/blob/0e99abf5422a5e2521e9c54c99f9bd7c7befb691/roles/vim/files/.vim/plugin/autocmnds.vim#L38


Should be able to swap out `Format` with `FormatWrite`

**Comment by ahmedelgabri** [comment]
I didn't know that `FormatWrite` existed…, thanks

**Comment by zlksnk** [comment]
So something like `autocmd BufWritePre *.ts,*.js,*.tsx,*.jsx,*.json Format` should work? This is in my case resulting in save to run first and then it formats second and so it leaves my buffer in an edited state.

**Comment by mhartington** [comment]
https://github.com/mhartington/formatter.nvim#format-on-save

**Comment by zlksnk** [comment]
Notice `41C` converting to `40C` by formatter

![Animation](https://user-images.githubusercontent.com/26769888/117672089-5a99c080-b15e-11eb-8905-31d8019f2d23.gif)

Is it possible to hold the write till buffer gets updated with the newly formatted text?

**Comment by mhartington** [comment]
Please open a new issue with a complete example and reproduction
